### DISCUSSION TITLE
Question Regarding Step Size Variations in Renewable Energy Device Simulations with ANDES

### DISCUSSION THREAD

**Discussion by cqss11** [CURENT/andes#489] [Q&A]
Hello Dr. Cui,

I am currently exploring the renewable energy device models in ANDES and I have come across a point of interest regarding the simulation step size. According to the ANDES documentation, it is noted that the step size can vary between 1/30 to 1/120 of a second with minimal impact on the results. Specifically, the documentation suggests that a step size of 1/120 sec is typically used, but a larger step size of 1/30 sec can yield almost the same outcomes.

Generally, a smaller simulation timestep will reveal more details in the simulation curves, especially in regions where the system's response is rapidly changing.

My query is, in scenarios where the step sizes differ significantly (for example, 0.01s and 0.0001s), have there been instances where the simulation curves exhibited notable differences? Understanding this would be crucial for my simulations where computational efficiency is as important a factor as accuracy.

Thank you for your guidance and for the exceptional work on ANDES！

Best regards,
cqss1

**Comment by cuihantao** [comment]
Hello,

the step size is chosen based on the timescales of system dynamics, which are prescribed by the differential equations and their parameters. The models in ANDES are in the transient stability timeframe, so that translates to tens of milliseconds or so. Both 1/30 or 1/120 sec fall in that range.

1/120 is more commonly used in commercial programs where a partitioned solution of diff. eq and algebraic equations are used. 1/30 gives similar results for implicit methods used in ANDES.

The automatic stepping algorithm in ANDES will automatically shrink the step size if convergence is not detected.

> have there been instances where the simulation curves exhibited notable differences

If your model involves very fast dynamics, such as electromagnetic transients, they will only be captured by the small step sizes. Otherwise, the results won't significantly differ.
